zerver/lib/actions.py

```diff
@@ -4923,22 +4924,30 @@ def notify_user_update_custom_profile_data(user_profile: UserProfile,
         field_value = ujson.dumps(field['value'])  # type: Union[int, str, List[int], None]
     else:
         field_value = field['value']
+    field_rendered_value = field['rendered_value']
     payload = dict(user_id=user_profile.id, custom_profile_field=dict(id=field['id'],
-                                                                      value=field_value))
+                                                                      value=field_value,
+                                                                      rendered_value=field_rendered_value))
     event = dict(type="realm_user", op="update", person=payload)
     send_event(user_profile.realm, event, active_user_ids(user_profile.realm.id))
 
 def do_update_user_custom_profile_data(user_profile: UserProfile,
                                        data: List[Dict[str, Union[int, str, List[int]]]]) -> None:
     with transaction.atomic():
         for field in data:
-            field_value, created = CustomProfileFieldValue.objects.update_or_create(
+            field_value, created = CustomProfileFieldValue.objects.get_or_create(
                 user_profile=user_profile,
-                field_id=field['id'],
-                defaults={'value': field['value']})
+                field_id=field['id'])
+            field_value.value = field['value']
+            if field_value.field.is_renderable():
+                field_value.rendered_value = bugdown_convert(str(field['value']))
```

Comment:
We should pass the `message_realm` argument to `bugdown_convert`; otherwise custom emoji won't work.  Might actually not be a bad idea to use a custom emoji in the test example to be sure this is working correctly.